data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/RetryConfiguration.java

```diff
@@ -11,29 +11,42 @@
 
 public class RetryConfiguration {
   public static final String DLQ_FILE = "dlq_file";
+  public static final String MAX_RETRIES = "max_retries";
 
   private final String dlqFile;
+  private final int maxRetries;
 
   public String getDlqFile() {
     return dlqFile;
   }
 
+  public int getMaxRetries() {
+    return maxRetries;
+  }
+
   public static class Builder {
     private String dlqFile;
+    private int maxRetries = Integer.MAX_VALUE;
 
     public Builder withDlqFile(final String dlqFile) {
       checkNotNull(dlqFile, "dlqFile cannot be null.");
       this.dlqFile = dlqFile;
       return this;
     }
 
+    public Builder withMaxRetries(final Integer maxRetries) {
```

Comment:
Can we provide more min and max validations around the maxRetries. There are some edge cases like negative numbers and `Integer.MAX_VALUE` that I am not convinced we should support.

Reply:
`Integer.MAX_VALUE` should be OK to support indefinitely retries.